### PR TITLE
moves finding split points into FATE

### PR DESCRIPTION
**Pull request by keith-turner**
1. Moves finding split points from tablet from a custom thread pool in the manager into FATE.  This is done by the new Repo named FindSplits.
 2. Stops tracking what splits are running in manager memory and moves this into FATE.  This is not implemented yet, plan to do that in another PR
 3. Stops tracking what tablets are unsplittable in manager memory. This should be tracked in metadata table per #4177.  This change can be in its own commit.  In this commit the in memory tracking is simply removed.

These changes will reduce manager memory usage. These changes also will make splitting tablets work much better if FATE is distributed for two reasons.  First it allows the computation to spread over multiple manager processes.  Second it allows tracking of what is splitting to move from memory to persisted storage and not rely on a map in a single process.

This is a WIP commit because it requires FATE changes.  The needed FATE changes were stubbed out in this PR and implemented with a in memory set that would need to be persisted.

### CODE REVIEW

**Review by cshannon — APPROVED** [review]
The changes here LGTM. The new FindSplits repo looks idempotent to me (it will just recompute the splits if fails) and I didn't see any other issues with the changes.

File: server/manager/src/main/java/org/apache/accumulo/manager/split/Splitter.java [hunk 1]
```diff
@@ -84,52 +74,34 @@ public int hashCode() {
...
-
   public Splitter(ServerContext context) {
-    this.splitExecutor = context.threadPools().createExecutorService(context.getConfiguration(),
-        Property.MANAGER_SPLIT_WORKER_THREADS, true);
+    int numThreads = context.getConfiguration().getCount(Property.MANAGER_SPLIT_WORKER_THREADS);
+    // Set up thread pool that constrains the amount of task it queues and when full discards task.
+    // The purpose of this is to avoid reading lots of data into memory if lots of tablets need to
+    // split.
+    BlockingQueue<Runnable> queue = new ArrayBlockingQueue<>(10000);
+    this.splitExecutor = context.threadPools().createThreadPool(numThreads, numThreads, 0,
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/split/Splitter.java:83]: You could remove the branch in ThreadPools.createExecutorService that handles the property MANAGER_SPLIT_WORKER_THREADS.

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/split/FindSplits.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      log.trace("Table {} no longer exist, so not gonna try to find a split point for it", extent);
+      return null;
+    }
+
+    if (tabletMetadata.getOperationId() != null) {
+      log.debug("Not splitting {} because it has operation id {}", tabletMetadata.getExtent(),
+          tabletMetadata.getOperationId());
+      return null;
+    }
+
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/split/FindSplits.java:60]: Do we need to add a check for walogs?

Comment (keith-turner) [reply server/manager/src/main/java/org/apache/accumulo/manager/tableOps/split/FindSplits.java:60]: checked in 0f1f2fa

File: server/manager/src/main/java/org/apache/accumulo/manager/split/Splitter.java (hunk 1 above)
Comment (keith-turner) [reply server/manager/src/main/java/org/apache/accumulo/manager/split/Splitter.java:83]: removed 0f1f2fa

**Pull request merged**